server/protocols/jmap/src/main/java/org/apache/james/jmap/methods/SetMessagesCreationProcessor.java

```diff
@@ -333,19 +338,48 @@ private MetaDataWithContent createMessageInOutbox(MessageWithId.CreationMessageE
     }
 
     private Flags getMessageFlags(CreationMessage message) {
+        Preconditions.checkState(!(hasKeywords(message) && isFlag(message)),
+                "Does not support both keywords and is*");
+        if (hasKeywords(message)) {
+            return buildFlagsFromKeywords(message.getKeywords().get());
+        } else {
+            return buildFlagsFromIsFlag(message);
+        }
+    }
+
+    private boolean isFlag(CreationMessage message) {
+        return message.isIsAnswered().isPresent()
+            || message.isIsDraft().isPresent()
+            || message.isIsFlagged().isPresent()
+            || message.isIsUnread().isPresent();
+    }
+
+    private boolean hasKeywords(CreationMessage message) {
+        return message.getKeywords().isPresent();
+    }
+
+    private Flags buildFlagsFromKeywords(ImmutableSet<JmapFlag> keywords) {
```

Comment:
JmapProcessor uses the MessageManager to manage message and leans on imap flags to append message
This method is same the bridge which convert from JmapFlags to ImapFlags (in this case is Flags).
So should we move the converting to lower layer (manager layer)?
Or should we define new ImapFlags (which leans on Flags) then Jmap won't care about javax.mail.Flags anymore?
Which one should I do?